src/main/scala/net/kaliber/play/prerender/PrerenderActionBuilders.scala

```diff
@@ -40,13 +40,25 @@ case class PrerenderActionBuilders(config: PrerenderConfig)(implicit ec: Executi
   }
 
   private def shouldBePrerendered(request: RequestHeader) =
-    config.enabled && isGetRequest(request) && isSearchEngineRequest(request)
+    config.enabled && isGetRequest(request) && isSearchEngineRequest(request) && hasOnlyAllowedParams(request) && !isExcludedWithRegex(request)
 
   private def isGetRequest(request: RequestHeader) = request.method == "GET"
 
   private def isSearchEngineRequest(request: RequestHeader) =
     isEscapedFragmentUrl(request) || hasSearchEngineUserAgent(request)
 
+  private def hasOnlyAllowedParams(request: RequestHeader) = {
+    val requestParams = request.queryString.keys.toSet
+    requestParams subsetOf config.limitedParams.getOrElse(requestParams)
+  }
+
+  private def isExcludedWithRegex(request: RequestHeader) = {
+    config.excludeRegex.getOrElse("").r.findFirstMatchIn(request.uri) match {
+      case None => false
+      case Some(matchingSubstring) => !matchingSubstring.toString().isEmpty
+    }
```

Comment:
That is a consequence of me wanting override the behavior of the regex when an empty string is used for the configuration. Normally, "" would match any part of the string, but for my implementation and for usability, I wanted this function to return false when "" was used as the regex. Perhaps it is better to just go with the default behavior, and assume that anything None is the only value that should be used to mean "don't exclude anything". I'll work on it today so you can see what you think.